pymc3_models/models/NaiveBayes.py

```diff
@@ -0,0 +1,283 @@
+        distribution of the parameters :math:`\\vec{\\mu}`, :math:`\\vec{\\pi}`
+        and :math:`\\vec{\\sigma}`. To compute the above probability, we need
+        to integrate over the values of these parameters:
+
+        .. math::
+
+            P(y=c|\\vec{x}, \\mathcal{D}) = \\left[\\int Cat(y=c|\\vec{\\pi})P(\\vec{\\pi}|\\
+                                            \\mathcal{D})\\mathrm{d}\\vec{\\pi}\\right]
+                                            \\int P(\\vec{x}|\\vec{\\mu}, \\vec{\\sigma})\\
+                                            P(\\vec{\\mu}|\\mathcal{D})\\
+                                            P(\\vec{\\sigma}|\\mathcal{D})\\
+                                            \\mathrm{d}\\vec{\\mu}\\mathrm{d}\\vec{\\sigma}
+
+        Parameters
+        ----------
+        X : numpy array, shape [num_training_samples, num_pred]. Contains the points
+            for which we want to predict the class
+
+        Returns
+        -------
+        A numpy array of shape [num_training_samples, num_cats] that contains the probabilities
+        that each sample belong to each category.
+
+        References
+        ----------
+        .. [1] Murphy, K. P. (2012). Machine learning: a probabilistic perspective.
+        """
+
+        if self.trace is None:
+            raise PyMC3ModelsError("Run fit() on the model before predict()")
+
+        predictions = np.array([])
+        for x in X:
+            prob_per_sample = scipy.stats.norm(self.trace['mu'], self.trace['sigma']).pdf(x)
+            prob_per_feature = [np.sum(prob_per_sample[:, :, i], axis=0)/len(self.trace['mu'])
+                                for i in range(self.num_pred)]
+            prob_per_class = normalize(ft.reduce(lambda x, y: x*y, prob_per_feature))
+            if len(predictions) == 0:
+                predictions = prob_per_class
+            else:
+                predictions = np.vstack((predictions, prob_per_class))
+
+        return predictions
+
+    def predict(self, X):
+        """
+        Classify new data with a trained Naive Bayes model.
+
+        Parameters
+        ----------
+        X : numpy array, shape [num_training_samples, num_pred]. Contains the data
+            to classify.
+
+        Returns
+        -------
+        A numpy array of shape [num_training_samples,] that contains the predicted class to
+        which the data points belong.
+        """
+        proba = self.predict_proba(X)
+        pred = np.argmax(proba, axis=1)
+        return pred
+
+    def score(self, X, y):
+        """
+        Scores new data with a trained model.
+
+        Parameters
+        ----------
+        X : numpy array, shape [num_training_samples, num_pred]. Contains the data points.
+
+        y : numpy array, shape [num_training_samples,]. Contains the category of the data points.
+
+        Returns
+        -------
+        A float representing the score of the predictions.
```

Comment:
Maybe put `representing the accuracy score` to be explicit about what scoring method you use.